lib/fog/compute/ecloud.rb

```diff
@@ -429,8 +429,8 @@ def canonicalize_resource(path)
           uri, query_string = path.split("?")
           return uri.downcase if query_string.nil?
           query_string_pairs = query_string.split("&").sort.map { |e| e.split("=") }
-          tm_query_string = query_string_pairs.map { |x| "#{x.first.downcase}:#{x.last}" }.join("\n")
-          "#{uri.downcase}\n#{tm_query_string}\n"
+          tm_query_string = query_string_pairs.map { |x| "#{x.first.downcase}:#{x.last.downcase}" }.join("\n")
+          "#{uri.downcase}\n#{tm_query_string}"
```

Comment:
Looking at API docs going back to 2015-05-01 it requires that keys and values be lowercased.

I'm not sure if it's possible to use an older or deprecated version in a private cloud instance. All of the private instances I have access to have deprecated 2015-05-01 and older.